The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.

[20250305.log](https://github.com/user-attachments/files/19082344/20250305.log)

first seen in spark-rapids-ml_nightly, run: 649 (note this run also failed https://github.com/NVIDIA/spark-rapids-ml/issues/858) in branch-25.02

the commit did not change, it looks like another non-determistic tests here
```
[2025-03-05T04:56:31.442Z] FAILED tests/test_logistic_regression.py::test_compat_sparse_multinomial[True] - pyspark.errors.exceptions.captured.PythonException: 
[2025-03-05T04:56:31.442Z]   An exception was thrown from the Python worker. Please see the stack trace below.
[2025-03-05T04:56:31.442Z] Traceback (most recent call last):
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.442Z]     process()
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.442Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.442Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.442Z]     for batch in iterator:
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.442Z]     for series in iterator:
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.442Z]     raise RuntimeError(
[2025-03-05T04:56:31.442Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.442Z] 
[2025-03-05T04:56:31.442Z] 
[2025-03-05T04:56:31.442Z] JVM stacktrace:
[2025-03-05T04:56:31.442Z] org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 590.0 failed 1 times, most recent failure: Lost task 0.0 in stage 590.0 (TID 1800) (cuml-build-jenkins-spark-rapids-ml-nightly-649-kgtzg-7lr5d executor driver): org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.442Z]     process()
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.442Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.442Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.442Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.442Z]     for batch in iterator:
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.443Z]     for series in iterator:
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.443Z]     raise RuntimeError(
[2025-03-05T04:56:31.443Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.443Z] 
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:561)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:118)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:102)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:514)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-03-05T04:56:31.443Z] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-03-05T04:56:31.443Z] 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:891)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:891)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:92)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:139)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
[2025-03-05T04:56:31.443Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2025-03-05T04:56:31.443Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2025-03-05T04:56:31.443Z] 	at java.lang.Thread.run(Thread.java:750)
[2025-03-05T04:56:31.443Z] 
[2025-03-05T04:56:31.443Z] Driver stacktrace:
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2790)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2726)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2725)
[2025-03-05T04:56:31.443Z] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-05T04:56:31.443Z] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-05T04:56:31.443Z] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2725)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.443Z] 	at scala.Option.foreach(Option.scala:407)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2989)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2928)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2917)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:976)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2258)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2279)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2298)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2323)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1022)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:408)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.collect(RDD.scala:1021)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:448)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:4038)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4208)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4206)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:118)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:195)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:103)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:65)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4206)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:4035)
[2025-03-05T04:56:31.443Z] 	at sun.reflect.GeneratedMethodAccessor118.invoke(Unknown Source)
[2025-03-05T04:56:31.443Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-03-05T04:56:31.443Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2025-03-05T04:56:31.443Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-03-05T04:56:31.443Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-05T04:56:31.443Z] 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-03-05T04:56:31.443Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-03-05T04:56:31.443Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-03-05T04:56:31.443Z] 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-05T04:56:31.443Z] 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-05T04:56:31.443Z] 	at java.lang.Thread.run(Thread.java:750)
[2025-03-05T04:56:31.443Z] Caused by: org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.443Z]     process()
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.443Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.443Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.443Z]     for batch in iterator:
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.443Z]     for series in iterator:
[2025-03-05T04:56:31.443Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.443Z]     raise RuntimeError(
[2025-03-05T04:56:31.443Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.443Z] 
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:561)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:118)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:102)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:514)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-03-05T04:56:31.443Z] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-03-05T04:56:31.443Z] 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:891)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:891)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2025-03-05T04:56:31.443Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:92)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:139)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
[2025-03-05T04:56:31.444Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2025-03-05T04:56:31.444Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2025-03-05T04:56:31.444Z] 	... 1 more
[2025-03-05T04:56:31.444Z] FAILED tests/test_logistic_regression.py::test_compat_sparse_multinomial[False] - pyspark.errors.exceptions.captured.PythonException: 
[2025-03-05T04:56:31.444Z]   An exception was thrown from the Python worker. Please see the stack trace below.
[2025-03-05T04:56:31.444Z] Traceback (most recent call last):
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.444Z]     process()
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.444Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.444Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.444Z]     for batch in iterator:
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.444Z]     for series in iterator:
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.444Z]     raise RuntimeError(
[2025-03-05T04:56:31.444Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.444Z] 
[2025-03-05T04:56:31.444Z] 
[2025-03-05T04:56:31.444Z] JVM stacktrace:
[2025-03-05T04:56:31.444Z] org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 607.0 failed 1 times, most recent failure: Lost task 0.0 in stage 607.0 (TID 1817) (cuml-build-jenkins-spark-rapids-ml-nightly-649-kgtzg-7lr5d executor driver): org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.444Z]     process()
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.444Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.444Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.444Z]     for batch in iterator:
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.444Z]     for series in iterator:
[2025-03-05T04:56:31.444Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.444Z]     raise RuntimeError(
[2025-03-05T04:56:31.444Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.444Z] 
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:561)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:118)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:102)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:514)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-03-05T04:56:31.444Z] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-03-05T04:56:31.444Z] 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:891)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:891)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:92)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:139)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
[2025-03-05T04:56:31.444Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2025-03-05T04:56:31.444Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2025-03-05T04:56:31.444Z] 	at java.lang.Thread.run(Thread.java:750)
[2025-03-05T04:56:31.444Z] 
[2025-03-05T04:56:31.444Z] Driver stacktrace:
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.failJobAndIndependentStages(DAGScheduler.scala:2790)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2(DAGScheduler.scala:2726)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$abortStage$2$adapted(DAGScheduler.scala:2725)
[2025-03-05T04:56:31.444Z] 	at scala.collection.mutable.ResizableArray.foreach(ResizableArray.scala:62)
[2025-03-05T04:56:31.444Z] 	at scala.collection.mutable.ResizableArray.foreach$(ResizableArray.scala:55)
[2025-03-05T04:56:31.444Z] 	at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:49)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:2725)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.$anonfun$handleTaskSetFailed$1$adapted(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.444Z] 	at scala.Option.foreach(Option.scala:407)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:1211)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.doOnReceive(DAGScheduler.scala:2989)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2928)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGSchedulerEventProcessLoop.onReceive(DAGScheduler.scala:2917)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.util.EventLoop$$anon$1.run(EventLoop.scala:49)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.scheduler.DAGScheduler.runJob(DAGScheduler.scala:976)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2258)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2279)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2298)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.SparkContext.runJob(SparkContext.scala:2323)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.$anonfun$collect$1(RDD.scala:1022)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:151)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDDOperationScope$.withScope(RDDOperationScope.scala:112)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.withScope(RDD.scala:408)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.rdd.RDD.collect(RDD.scala:1021)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SparkPlan.executeCollect(SparkPlan.scala:448)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.Dataset.$anonfun$collectToPython$1(Dataset.scala:4038)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.Dataset.$anonfun$withAction$2(Dataset.scala:4208)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.QueryExecution$.withInternalError(QueryExecution.scala:526)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.Dataset.$anonfun$withAction$1(Dataset.scala:4206)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$6(SQLExecution.scala:118)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SQLExecution$.withSQLConfPropagated(SQLExecution.scala:195)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SQLExecution$.$anonfun$withNewExecutionId$1(SQLExecution.scala:103)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.SparkSession.withActive(SparkSession.scala:827)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.execution.SQLExecution$.withNewExecutionId(SQLExecution.scala:65)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.Dataset.withAction(Dataset.scala:4206)
[2025-03-05T04:56:31.444Z] 	at org.apache.spark.sql.Dataset.collectToPython(Dataset.scala:4035)
[2025-03-05T04:56:31.444Z] 	at sun.reflect.GeneratedMethodAccessor118.invoke(Unknown Source)
[2025-03-05T04:56:31.444Z] 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2025-03-05T04:56:31.444Z] 	at java.lang.reflect.Method.invoke(Method.java:498)
[2025-03-05T04:56:31.445Z] 	at py4j.reflection.MethodInvoker.invoke(MethodInvoker.java:244)
[2025-03-05T04:56:31.445Z] 	at py4j.reflection.ReflectionEngine.invoke(ReflectionEngine.java:374)
[2025-03-05T04:56:31.445Z] 	at py4j.Gateway.invoke(Gateway.java:282)
[2025-03-05T04:56:31.445Z] 	at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
[2025-03-05T04:56:31.445Z] 	at py4j.commands.CallCommand.execute(CallCommand.java:79)
[2025-03-05T04:56:31.445Z] 	at py4j.ClientServerConnection.waitForCommands(ClientServerConnection.java:182)
[2025-03-05T04:56:31.445Z] 	at py4j.ClientServerConnection.run(ClientServerConnection.java:106)
[2025-03-05T04:56:31.445Z] 	at java.lang.Thread.run(Thread.java:750)
[2025-03-05T04:56:31.445Z] Caused by: org.apache.spark.api.python.PythonException: Traceback (most recent call last):
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 830, in main
[2025-03-05T04:56:31.445Z]     process()
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 822, in process
[2025-03-05T04:56:31.445Z]     serializer.dump_stream(out_iter, outfile)
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 345, in dump_stream
[2025-03-05T04:56:31.445Z]     return ArrowStreamSerializer.dump_stream(self, init_stream_yield_batches(), stream)
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 86, in dump_stream
[2025-03-05T04:56:31.445Z]     for batch in iterator:
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/sql/pandas/serializers.py", line 338, in init_stream_yield_batches
[2025-03-05T04:56:31.445Z]     for series in iterator:
[2025-03-05T04:56:31.445Z]   File "/root/miniconda3/lib/python3.10/site-packages/pyspark/python/lib/pyspark.zip/pyspark/worker.py", line 540, in func
[2025-03-05T04:56:31.445Z]     raise RuntimeError(
[2025-03-05T04:56:31.445Z] RuntimeError: The length of output in Scalar iterator pandas UDF should be the same with the input's; however, the length of output was 3 and the length of input was 4.
[2025-03-05T04:56:31.445Z] 
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.handlePythonException(PythonRunner.scala:561)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:118)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.execution.python.PythonArrowOutput$$anon$1.read(PythonArrowOutput.scala:102)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.api.python.BasePythonRunner$ReaderIterator.hasNext(PythonRunner.scala:514)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.InterruptibleIterator.hasNext(InterruptibleIterator.scala:37)
[2025-03-05T04:56:31.445Z] 	at scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:491)
[2025-03-05T04:56:31.445Z] 	at scala.collection.Iterator$$anon$10.hasNext(Iterator.scala:460)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.catalyst.expressions.GeneratedClass$GeneratedIteratorForCodegenStage2.processNext(Unknown Source)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.execution.BufferedRowIterator.hasNext(BufferedRowIterator.java:43)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.execution.WholeStageCodegenExec$$anon$1.hasNext(WholeStageCodegenExec.scala:760)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.sql.execution.SparkPlan.$anonfun$getByteArrayRdd$1(SparkPlan.scala:388)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2(RDD.scala:891)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.rdd.RDD.$anonfun$mapPartitionsInternal$2$adapted(RDD.scala:891)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.rdd.MapPartitionsRDD.compute(MapPartitionsRDD.scala:52)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.rdd.RDD.computeOrReadCheckpoint(RDD.scala:367)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.rdd.RDD.iterator(RDD.scala:331)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.scheduler.ResultTask.runTask(ResultTask.scala:92)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.TaskContext.runTaskWithListeners(TaskContext.scala:161)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.scheduler.Task.run(Task.scala:139)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:554)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1529)
[2025-03-05T04:56:31.445Z] 	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:557)
[2025-03-05T04:56:31.445Z] 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
[2025-03-05T04:56:31.445Z] 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
[2025-03-05T04:56:31.445Z] 	... 1 more
...
67 failed, 624 passed, 2 skipped, 107741 warnings in 3746.11s (1:02:26)
```

```
[2025-03-05T03:34:39.392Z] +-----------------------------------------------------------------------------------------+
[2025-03-05T03:34:39.392Z] | NVIDIA-SMI 570.86.10              Driver Version: 570.86.10      CUDA Version: 12.8     |
[2025-03-05T03:34:39.392Z] |-----------------------------------------+------------------------+----------------------+
[2025-03-05T03:34:39.392Z] | GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
[2025-03-05T03:34:39.392Z] | Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
[2025-03-05T03:34:39.392Z] |                                         |                        |               MIG M. |
[2025-03-05T03:34:39.392Z] |=========================================+========================+======================|
[2025-03-05T03:34:39.392Z] |   0  NVIDIA A100-PCIE-40GB          On  |   00000000:65:00.0 Off |                    0 |
[2025-03-05T03:34:39.392Z] | N/A   32C    P0             33W /  250W |       1MiB /  40960MiB |      0%      Default |
[2025-03-05T03:34:39.392Z] |                                         |                        |             Disabled |
[2025-03-05T03:34:39.392Z] +-----------------------------------------+------------------------+----------------------+
[2025-03-05T03:34:39.392Z]                                                                                          
[2025-03-05T03:34:39.392Z] +-----------------------------------------------------------------------------------------+
[2025-03-05T03:34:39.392Z] | Processes:                                                                              |
[2025-03-05T03:34:39.392Z] |  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
[2025-03-05T03:34:39.392Z] |        ID   ID                                                               Usage      |
[2025-03-05T03:34:39.392Z] |=========================================================================================|
[2025-03-05T03:34:39.392Z] |  No running processes found                                                             |
[2025-03-05T03:34:39.392Z] +-----------------------------------------------------------------------------------------+
```

--- a/python/src/spark_rapids_ml/classification.py
+++ b/python/src/spark_rapids_ml/classification.py
@@ -1451,7 +1451,7 @@ class LogisticRegressionModel(
 
                 data = {}
 
-                scores = lr.decision_function(pdf).T
+                scores = lr.decision_function(pdf)
                 assert isinstance(scores, cp.ndarray)
                 _num_classes = max(scores.shape[1] if len(scores.shape) == 2 else 2, 2)
 

--- a/python/src/spark_rapids_ml/clustering.py
+++ b/python/src/spark_rapids_ml/clustering.py
@@ -1,5 +1,5 @@
 #
-# Copyright (c) 2022-2024, NVIDIA CORPORATION.
+# Copyright (c) 2022-2025, NVIDIA CORPORATION.
 #
 # Licensed under the Apache License, Version 2.0 (the "License");
 # you may not use this file except in compliance with the License.

--- a/python/src/spark_rapids_ml/clustering.py
+++ b/python/src/spark_rapids_ml/clustering.py
@@ -115,7 +115,7 @@ class KMeansClass(_CumlClass):
             "verbose": False,
             "random_state": 1,
             "init": "scalable-k-means++",
-            "n_init": 1,
+            "n_init": "warn",  # See https://github.com/rapidsai/cuml/pull/6142 - this needs to be updated to "auto" for cuml 25.04
             "oversampling_factor": 2.0,
             "max_samples_per_batch": 32768,
         }

